sync3/extensions/extensions.go

```diff
@@ -77,6 +81,17 @@ func (r *Core) OnlyRooms() []string {
 	return r.Rooms
 }
 
+func (r *Core) InterpretAsInitial() {
+	// An omitted/nil value for lists and rooms normally means "no change".
+	// If this extension has never been specified before, nil means "all lists/rooms".
+	if r.Lists == nil {
+		r.Lists = []string{"*"}
+	}
+	if r.Rooms == nil {
+		r.Rooms = []string{"*"}
+	}
+}
+
```

Comment:
I did try to make GenericRequest define a method `Default` which returned a GenericRequest, and then do `Default().ApplyDelta(incomingRequest)` instead of `incomingRequest.InterpretAsInitial`. Unfortunately I couldn't get this working so deemed this good enough.